src/actions/arcActions.ts

```diff
@@ -452,8 +452,11 @@ export function createProposal(daoAvatarAddress : string, title : string, descri
     dispatch({ type: arcConstants.ARC_CREATE_PROPOSAL_PENDING, payload: null });
     try {
       const web3 : Web3 = Arc.Utils.getWeb3();
-      const ethAccountAddress : string = getState().web3.ethAccountAddress;
 
+      if(!web3.isAddress(beneficiary)) throw new Error(`Invalid address`);
```

Comment:
how does this error get caught and handled on the front end?

Reply:
It's caught and dispatched as a `ARC_CREATE_PROPOSAL_REJECTED`.
How would you like to handle it?

Reply:
ok, that works, i mean more that an error should be displayed in the UI

Reply:
update: https://github.com/daostack/alchemy/pull/18/commits/005286a25b051811ed007e726f8871c8e309c778

Moved validation to form itself.